src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -1125,6 +1300,334 @@ void ov::npuw::JustInferRequest::unsafe_during(std::size_t real_idx, std::size_t
+    // SECTION 2: Input/Output Tensor Extraction
+    // ================================================================================================
+
+    const auto& hfa_inputs = m_hfa_io[idx].inputs;
+    const auto& hfa_outputs = m_hfa_io[idx].outputs;
+    const auto& sdpa_info = hfa_desc._sdpa_attention_info;
+
+    // Use pre-cached SDPA indices
+    const auto& sdpa_in = sdpa_info._sdpa_indices;
+
+    auto past_key_tensor = hfa_inputs.at(sdpa_in.past_key);
+    auto past_value_tensor = hfa_inputs.at(sdpa_in.past_value);
+    auto query_tensor = hfa_inputs.at(sdpa_in.query);
+    auto present_key_tensor = hfa_inputs.at(sdpa_in.present_key);
+    auto attention_mask_tensor = hfa_inputs.at(sdpa_in.attention_mask);
+    auto present_value_tensor = hfa_inputs.at(sdpa_in.present_value);
+
+    auto attention_output_tensor = hfa_outputs.at(0);
+
+    // ================================================================================================
+    // SECTION 3: State Initialization
+    // ================================================================================================
+
+    // Get tile infer requests
+    auto& regular_tile_request =
+        comp_model_desc.hfa_infer_requests[CompiledModel::CompiledModelDesc::HFATileIdx::REGULAR_TILE];
+    auto& final_tile_request =
+        comp_model_desc.hfa_infer_requests[CompiledModel::CompiledModelDesc::HFATileIdx::FINAL_TILE];
+
+    // Use pre-cached indices (populated during compilation)
+    const auto& tile_in = sdpa_info._tile_input_indices;
+    const auto& tile_out = sdpa_info._tile_output_indices;
+
+    // Initialize state tensors to zero/negative infinity
+    auto state_acc = regular_tile_request->get_tensor(hfa_desc._compiled_tile_model->inputs()[tile_in.acc]);
+    auto state_max = regular_tile_request->get_tensor(hfa_desc._compiled_tile_model->inputs()[tile_in.max]);
+    auto state_sum = regular_tile_request->get_tensor(hfa_desc._compiled_tile_model->inputs()[tile_in.d]);
+
+    // Template lambda for state initialization
+    auto initialize_state = [](auto& acc, auto& max, auto& sum, ov::element::Type type) {
+        if (type == ov::element::f16) {
+            std::fill_n(acc->template data<ov::float16>(), acc->get_size(), ov::float16(0.0f));
+            std::fill_n(max->template data<ov::float16>(), max->get_size(), ov::float16(-65500.0f));
+            std::fill_n(sum->template data<ov::float16>(), sum->get_size(), ov::float16(0.0f));
+        } else if (type == ov::element::f32) {
+            std::fill_n(acc->template data<float>(), acc->get_size(), 0.0f);
+            std::fill_n(max->template data<float>(), max->get_size(), -65500.0f);
+            std::fill_n(sum->template data<float>(), sum->get_size(), 0.0f);
+        } else {
+            OPENVINO_THROW("Unsupported element type for HFA state tensors: ", type);
+        }
+    };
+
+    initialize_state(state_acc, state_max, state_sum, state_acc->get_element_type());
```

Comment:
is it for every infer we need to reset states?  looks not optimal but i think swapping zero tensors and zeoring them in advance in say async - wouldn't make sense either